### PR TITLE
Add and test for missing method overrides in FabricBlockSettings

### PR DESCRIPTION
**Pull request by modmuss50**
5 mins into porting my mod to 1.20, I found this problem. Its amazing what you find when you actually use something 😆 

This should be a none breaking change, also added a basic test to ensure it doesnt get like this again.

### CODE REVIEW

**Comment by haykam821** [comment]
Could these tests be expanded to `FabricItemSettings` as well? There might be other classes that could benefit from these changes but item settings seem like the most prone to changes.

**Comment by modmuss50** [comment]
> Could these tests be expanded to FabricItemSettings as well?

Yes, good idea. Its kinda a pain as its in a diffrent module. I may just copy/paste the tests over to the other module 🤔

**Comment by Technici4n** [comment]
Ideally the whole thing would be replaced by an injected interface

File: fabric-object-builder-api-v1/src/testmod/java/net/fabricmc/fabric/test/object/builder/FabricBlockSettingsTest.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+		}
+
+		throw new IllegalStateException("Missing method overrides in FabricBlockSettings: " + String.join(", ", missingMethods));
+	}
+
+	private List<String> getMethods(Class<?> clazz) {
+		List<String> methods = new ArrayList<>();
+
+		for (final Method method : clazz.getDeclaredMethods()) {
+			if (method.getReturnType() != clazz) {
```
Comment (liach) [inline fabric-object-builder-api-v1/src/testmod/java/net/fabricmc/fabric/test/object/builder/FabricBlockSettingsTest.java]: I would recommend just checking `getMethods()`, and report any non-bridge method that returns a vanilla `Settings`.

Comment (modmuss50) [reply fabric-object-builder-api-v1/src/testmod/java/net/fabricmc/fabric/test/object/builder/FabricBlockSettingsTest.java]: Good idea, its much cleaner now 👍

**Pull request merged**